lib/relation-definition.js

```diff
@@ -1714,6 +1785,19 @@ HasOne.prototype.related = function (refresh, params) {
   }
 };
 
+/**
+ * Define a Promise-based method for the hasOne relation itself
+ * - order.customer.get(refresh): Load the target model instance asynchronously
+ *
+ * @param {Boolean} refresh Reload from the data source
+ * @returns {Promise}
+ */
+
+HasOne.prototype.get = function (refresh) {
+  var cb = utils.createPromiseCallback();
+  return this.related(refresh, cb);
```

Comment:
I guess we can mark the failing test with `it.skip` for now and investigate later. Unless @raymondfeng has time to help you with this now?